core/src/main/scala/tech/beshu/ror/accesscontrol/logging/AuditRequestContextBasedOnAclResult.scala

```diff
@@ -65,18 +74,35 @@ class AuditRequestContextBasedOnAclResult[B <: BlockContext](requestContext: Req
   override val `type`: String = requestContext.`type`.value
   override val taskId: Long = requestContext.taskId
   override val httpMethod: String = requestContext.method.m
-  override val loggedInUserName: Option[String] = blockContext.flatMap(_.userMetadata.loggedUser.map(_.id.value.value))
+  override val loggedInUserName: Option[String] = maybeLoggedUser.map(_.id.value.value)
   override val impersonatedByUserName: Option[String] =
-    blockContext
-      .flatMap(_.userMetadata.loggedUser)
+    maybeLoggedUser
       .flatMap {
         case DirectlyLoggedUser(_) => None
         case ImpersonatedUser(_, impersonatedBy) => Some(impersonatedBy.value.value)
       }
-  override val involvesIndices: Boolean = blockContext match {
-    case Some(_: GeneralIndexRequestBlockContext | _: TemplateRequestBlockContext | _: SnapshotRequestBlockContext) => true
-    case _ => false
-  }
+
   override val attemptedUserName: Option[String] = requestContext.basicAuth.map(_.credentials.user.value.value)
   override val rawAuthHeader: Option[String] = requestContext.rawAuthHeader.map(_.value.value)
 }
+
+object AuditRequestContextBasedOnAclResult {
```

Comment:
I'd rather:
* create extension method for a block context
* change the name, because `inspect indices` are too general
* move the class to BlockContext file